Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.mm

```diff
@@ -185,17 +192,23 @@
             return;
         // This logic is from CrossfadeGeneratedImage.h, but we copy it to avoid some overhead and
         // also because that class is not supposed to be used in GPUP.
-        // FIXME: as above, not using context().platformContext() here is likely dubious.
+        // FIXME: As above, not using context().platformContext() here is likely dubious.
         trackImage->context().setAlpha(1.0f - progress);
         trackImage->context().drawConsumingImageBuffer(WTFMove(fromImage), IntPoint(), ImagePaintingOptions { CompositeOperator::SourceOver });
         trackImage->context().setAlpha(progress);
         trackImage->context().drawConsumingImageBuffer(WTFMove(toImage), IntPoint(), ImagePaintingOptions { CompositeOperator::PlusLighter });
     }
-    context.drawConsumingImageBuffer(WTFMove(trackImage), inflatedTrackRect.location());
+
+    {
+        GraphicsContextStateSaver rotationStateSaver(context);
+        if (isVertical)
+            SwitchMacUtilities::rotateContextForVerticalWritingMode(context, inflatedTrackRect.location(), inflatedTrackRect.height());
+        context.drawConsumingImageBuffer(WTFMove(trackImage), inflatedTrackRect.location());
+    }
 
     if (isFocused) {
         auto color = colorFromCocoaColor([NSColor keyboardFocusIndicatorColor]).opaqueColor();
-        context.drawFocusRing(Vector { trackRect }, 0, 1.0f, color);
+        context.drawFocusRing(Vector { logicalTrackRect }, 0, 0, color);
```

Comment:
Isn't `logicalTrackRect` really a physical rect, given the lack of rotation needed here?